keras/src/saving/orbax_util.py

```diff
@@ -0,0 +1,137 @@
+            "configuration. Cannot recreate model from checkpoint. This "
+            "may happen when saving weights only."
+        )
+
+    # Recreate model from config
+    model_config = loaded_state["model_config"]
+
+    # Determine model type from config
+    if "layers" in model_config:
+        # Sequential model
+        from keras.src.models import sequential as sequential_module
+
+        model = sequential_module.Sequential.from_config(
+            model_config, custom_objects=custom_objects
+        )
+    else:
+        # Functional model
+        model = model_module.Model.from_config(
+            model_config, custom_objects=custom_objects
+        )
+
+    # Compile if requested and if the original model was compiled
+    # (we can infer this from the presence of optimizer_variables)
+    if compile and "optimizer_variables" in loaded_state:
+        # Try to compile with default settings
+        # This may not work if the model was compiled with custom settings
+        try:
+            model.compile(optimizer="adam", loss="mse", metrics=["mae"])
+        except Exception:
+            # If compilation fails, leave the model uncompiled
+            pass
+
+    # Set the state in the model, but only for components that exist
+    state_to_set = {}
+
+    # Always load trainable and non-trainable variables
+    if "trainable_variables" in loaded_state:
+        state_to_set["trainable_variables"] = loaded_state[
+            "trainable_variables"
+        ]
+    if "non_trainable_variables" in loaded_state:
+        state_to_set["non_trainable_variables"] = loaded_state[
+            "non_trainable_variables"
+        ]
+
+    # Only load optimizer state if the model has an optimizer
+    if (
+        "optimizer_variables" in loaded_state
+        and hasattr(model, "optimizer")
+        and model.optimizer is not None
+    ):
+        # Ensure optimizer variables are created by doing a dummy
+        # apply_gradients. This creates the momentum/velocity
+        # variables that are needed
+        import numpy as np
+
+        # Create zero gradients for all trainable variables
+        zero_grads = [
+            backend.convert_to_tensor(np.zeros_like(v.numpy()))
+            for v in model.trainable_variables
+        ]
+        # Apply gradients to create optimizer slots
+        model.optimizer.apply_gradients(
+            zip(zero_grads, model.trainable_variables)
+        )
+        state_to_set["optimizer_variables"] = loaded_state[
+            "optimizer_variables"
+        ]
+
+    # Only load metrics state if the model has metrics variables
+    if (
+        "metrics_variables" in loaded_state
+        and hasattr(model, "metrics_variables")
+        and model.metrics_variables
+    ):
+        state_to_set["metrics_variables"] = loaded_state["metrics_variables"]
+
+    model.set_state_tree(state_to_set)
+
+    return model
+
+
+def _is_orbax_checkpoint(filepath):
```

Comment:
To follow the standard for utils, remove underscore at the beginning: `is_orbax_checkpoint`